### ISSUE TITLE
3.7.1 REGRESSION: Linking fails when building for aarch64 Windows

### ISSUE THREAD

**Issue by jakepetroules** [apple/swift-crypto#262]
### New Issue Checklist

- [x] I'm using the latest version of Swift Crypto (`main` branch) (_using 3.7.1 tag, but no changes since then are relevant_)
- [x] I read the [Contribution Guidelines](https://github.com/apple/swift-crypto/blob/master/CONTRIBUTING.md)
- [x] I searched for [existing GitHub issues](https://github.com/apple/swift-crypto/issues)

### Expected behavior
Links successfully as it did on 3.7.0.

### Actual behavior
```
lld-link: error: duplicate symbol: CCryptoBoringSSL_bn_add_words
>>> defined at E:\\test\\.build\\aarch64-unknown-windows-msvc\\debug\\CCryptoBoringSSL.build\\crypto\\fipsmodule\\bcm.c.o
>>> defined at E:\\test\\.build\\aarch64-unknown-windows-msvc\\debug\\CCryptoBoringSSL.build\\gen\\bcm\\bn-armv8-win.S.o

lld-link: error: duplicate symbol: CCryptoBoringSSL_bn_sub_words
>>> defined at E:\\test\\.build\\aarch64-unknown-windows-msvc\\debug\\CCryptoBoringSSL.build\\crypto\\fipsmodule\\bcm.c.o
>>> defined at E:\\test\\.build\\aarch64-unknown-windows-msvc\\debug\\CCryptoBoringSSL.build\\gen\\bcm\\bn-armv8-win.S.o
clang: error: linker command failed with exit code 1 (use -v to see invocation)
[464/465] Linking E:\test\.build\aarch64-unknown-windows-msvc\debug\test.exe
```

### Steps to reproduce

1. Create a new test package:

```swift
// swift-tools-version: 6.0
import PackageDescription

let package = Package(
    name: "test",
    products: [
        .executable(name: "test", targets: ["test"]),
    ],
    dependencies: [
        .package(url: "https://github.com/apple/swift-crypto.git", "2.0.0"..<"4.0.0"),
    ],
    targets: [
        .executableTarget(
            name: "test", dependencies: [.product(name: "Crypto", package: "swift-crypto")]),
    ]
)
```

2. Build using `swift build --triple aarch64-unknown-windows-msvc`

### If possible, minimal yet complete reproducer code (or URL to code)

(see above)

### Swift Crypto version/commit hash

3.7.1

### Environment

Swift version 6.0 (swift-6.0-RELEASE)
Target: aarch64-unknown-windows-msvc
Microsoft Windows [Version 10.0.22631.4169]

**Comment by jakepetroules** [comment]
This is a regression from #255

**Comment by Lukasa** [comment]
Ah dangit, this is what comes of not having a Windows CI. I _think_ I know what a fix would be. Can you try deleting this block and rebuilding? https://github.com/apple/swift-crypto/blob/35703579f63c2518fc929a1ce49805ba6134137c/Sources/CCryptoBoringSSL/include/CCryptoBoringSSL_base.h#L55-L60

**Comment by jakepetroules** [comment]
I can confirm that commenting that out resolves the build error. Adding `Platform.windows` to the line `.define("OPENSSL_NO_ASM", .when(platforms: [Platform.wasi])),` in Package.swift also works.

But presumably this is some upstream bug and we actually want Windows to be able to use the ASM implementation for better performance on x86_64 and arm64?

I did a cursory scan of commits since the 3.7.1 vendoring on https://boringssl.googlesource.com/boringssl/+log/refs/heads/master and nothing stood out.

**Comment by Lukasa** [comment]
It’s not an upstream bug. We add those lines, as an artefact of the way we interface with BoringSSL’s build system. 

Specifically, historically the assembly code portions of BoringSSL were templated out using a slightly arcane system called perlasm. We had to run that as part of our vendoring operation, as SwiftPM cannot do so. We had no CI for some platforms so I never felt confident about not regressing that.

However, more recent boringssls no longer need that step. This is good, but it turns out my concern about not having CI was founded: we didn’t notice that our settings now caused the symbols to be emitted twice. The fix is to unwind the forcing of these settings at our abstraction layer.

**Comment by gregcotten** [comment]
Just discovered this bug and relieved someone else ran into it! Any idea when a fix will land for this?

**Comment by Lukasa** [comment]
Hi @gregcotten, I think we're just waiting on someone getting time to make the change in the code. I can try to get this done now.

**Comment by Lukasa** [comment]
Mind testing local `main` @jakepetroules?

**Comment by gregcotten** [comment]
@Lukasa I can confirm using `main` fixes these linker issues for `aarch64-unknown-windows-msvc`
